galileo/src/render/render_bundle/mod.rs

```diff
@@ -94,8 +116,16 @@ impl RenderBundle {
         P: CartesianPoint3d<Num = N>,
     {
         if attach_to_map {
-            self.world_set.add_label(position, text, style, offset);
-        } else if let Some(set) = ScreenRenderSet::new_from_label(position, text, style, offset) {
+            // Apply DPI scaling to font size if set
+            let mut scaled_style = style.clone();
+            if self.dpi_scale_factor > 0.0 && self.dpi_scale_factor != 1.0 {
+                scaled_style.font_size *= self.dpi_scale_factor;
+            }
+            self.world_set
+                .add_label(position, text, &scaled_style, offset, self.dpi_scale_factor);
```

Comment:
It seems strange that you first create a scaled style, and then also pass `dpi_scale_factor` to the text service. I believe you should only do the second.